### PR TITLE
Added delete endpoint to remove application from mongodb

### PR DESCRIPTION
**Pull request by ttingle-ch**
__Short description outlining key changes/additions__
Added delete endpoint
__JIRA Ticket Number__
IDVA5-851
### Type of change

* [ ] Bug fix
* [x] New feature
* [ ] Breaking change
* [ ] Infrastructure change

### Pull request checklist

* [x] I have added unit tests for new code that I have added
* [ ] I have added/updated functional tests where appropriate
* [x] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/java.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/companieshouse/styleguides/blob/master/java_review.md#developer-actions-prior-to-code-commitreview-started)__

### CODE REVIEW

**Review by swalke-ch — DISMISSED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/java/uk/gov/companieshouse/acsp/controllers/AcspController.java [hunk 1]
```diff
@@ -81,4 +81,11 @@ public ResponseEntity<Object> checkHasApplication(@PathVariable("id") String id,
         return acspService.getAcspApplicationCount(id);
     }
 
+    @DeleteMapping("/acsp-api/user/{id}/application")
+    public ResponseEntity<Object> deleteApplication(@PathVariable("id") String id,
+                                                      @RequestHeader(value = ERIC_ACCESS_TOKEN) String requestId){
```
Comment (sampankumar) [inline src/main/java/uk/gov/companieshouse/acsp/controllers/AcspController.java:86]: Do you need this requestId header? Its automatically sent actually.

File: src/test/java/uk/gov/companieshouse/acsp/service/AcspServiceTest.java [hunk 2]
```diff
@@ -16,14 +17,12 @@
...
 import uk.gov.companieshouse.acsp.mapper.ACSPRegDataDtoDaoMapper;
-
 import java.util.Optional;
 
 import static org.junit.jupiter.api.Assertions.*;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.ArgumentMatchers.eq;
-import static org.mockito.Mockito.doNothing;
-import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.*;
```
Comment (sampankumar) [inline src/test/java/uk/gov/companieshouse/acsp/service/AcspServiceTest.java:25]: Use specific imports please.

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by hemerem-ch — APPROVED** [review]
LGTM

**Review by sampankumar — APPROVED** [review]

**Review by swalke-ch — APPROVED** [review]

**Pull request merged**